### PR TITLE
DebounceButton A/C switch

### PR DESCRIPTION
**Pull request by chuckwagoncomputing**

### CODE REVIEW

**Review by rusefillc — CHANGES_REQUESTED** [review]
we need mocked digital switches please nothing fancy?

**Review by rusefillc — APPROVED** [review]

**Comment by rusefillc** [comment]
fails multiple checks :(

**Review by mck1117 — APPROVED** [review]
new digital mocking looks good to me!

**Comment by rusefillc** [comment]
are we expanding scope before getting a green build of an OK version?

**Comment by rusefillc** [comment]
please build locally? it's not that I am bothered by emails but you itself would be more efficient if you are not waiting for CI to fail? is not it quicker locally?

File: firmware/hw_layer/debounce.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include "pin_repository.h"
 #include "engine_configuration.h"
 #include "hardware.h"
+#include "hal.h"
```
Comment (rusefillc) [inline firmware/hw_layer/debounce.cpp:12]: this does not look right. no one is including hal.h why would debounce.cpp?!

**Review by rusefillc — CHANGES_REQUESTED** [review]

File: firmware/hw_layer/debounce.cpp (hunk 1 above)
Comment (chuckwagoncomputing) [reply firmware/hw_layer/debounce.cpp]: brute force programming. it's removed now

Comment (rusefillc) [reply firmware/hw_layer/debounce.cpp]: I am not the one to criticize brute force programming.

**Comment by chuckwagoncomputing** [comment]
> please build locally? it's not that I am bothered by emails but you itself would be more efficient if you are not waiting for CI to fail? is not it quicker locally?

I am building locally. I push then pull in a different copy, so I don't accidentally push config changes etc.

**Comment by chuckwagoncomputing** [comment]
> are we expanding scope before getting a green build of an OK version?

Well, we expanded the scope from using DebounceButton for the A/C switch, to adding digital pin mocking so we can test it.
Beyond that, no. My initial implementation was broken. It was in an #if EFI_PROD_CODE, and when I got it out I discovered that it didn't actually work, because it didn't have access to the 'engine' object.

**Pull request merged**